### ISSUE TITLE
Move taglib from QHiMDTransfer into libhimd, remove id3tag dependency

### ISSUE THREAD

**Issue by thp** [linux-minidisc/linux-minidisc#20]
As discussed in #12, right now different ID3 tag libraries are used, and for upload and download and GUI and CLI (and combinations of those), ID3 tags are not always written or read.

To fix this, use taglib for reading and writing tags, and move the reading of tags (when transferring MP3s from PC to Hi-MD) as well as writing of tags (when transferring MP3s from Hi-MD to PC) directly from within libhimd, so that the MP3 writing/reading is consistent between CLI and GUI.

**Comment by karcherm** [comment]
I know that in #12 people agreed that tag writing should be moved to libhimd. I don't agree with that oppion for two reasons, though. libhimd is about accessing music on Hi-MD, not about accessing MP3 files. Furthermore, referencing a C++ library from libhimd "pollutes" not only libhimd forces applications that use libhimd to be aware of the fact that a C++ library got indirectly pulled when selecting how to link the application. On Linux, you need to call the linker via g++ instead of gcc if any static library you link is written in C++ (because gcc doesn't pull libstdc++), but it might even be relevant for dynamic libraries (that pull their dependencies themselves) if g++ uses different startup code than gcc (I am not sure if it does currently).
On the other hand, I am not against unifying code between himdcli and QHiMDTransfer. If we want himdcli to be more than a simple demonstration how one might use libhimd (this is how it started), we may certainly use C++ code in himdcli. libhimd on the other hand was originally meant to be low level enough to be used for example as an input plugin inside ffmpeg. I don't think they would be happy about a dependency on _any_ tag library. If we give up the goal of a minimal library for himd _access_, and have the new goal of creating a library for himd _music exchange_ instead, having a tag library dependency does not really matter. In that case, a long-term goal could be to also drop glib and rewrite the stuff based on the C++ standard library and/or Qt.

**Comment by thp** [comment]
Makes sense. What about creating a separate `libhimdtransfer` that depends on `libhimd` and includes the file transfer features (MP3 tagging, WAV file writing, etc...) and is depended upon by `QHiMDTransfer` and `himdcli` so that those don't need to include separate code?

In that case, I'd even suggest moving the `libmad` dependency and `mp3tools.c` out of libhimd and into `libhimdtransfer`.
